gossipd/gossip.c

```diff
@@ -1011,6 +1011,31 @@ static struct io_plan *hand_back_peer(struct io_conn *conn,
 			  read_returning_gossipfd, rpeer);
 }
 
+static struct io_plan *disconnect_peer(struct io_conn *conn, struct daemon *daemon,
+				       const u8 *msg)
+{
+	struct pubkey id;
+ 	struct peer *peer;
+
+	if (!fromwire_gossipctl_peer_disconnect(msg, &id))
+		master_badmsg(WIRE_GOSSIPCTL_PEER_DISCONNECT, msg);
+
+	peer = find_peer(daemon, &id);
+	if (peer && peer->local) {
+		/* This peer is local to this (gossipd) dameon */
+		io_close(peer->local->conn);
+		msg = towire_gossipctl_peer_disconnect_reply(msg);
+		daemon_conn_send(&daemon->master, take(msg));
+	} else {
+		status_trace("disconnect_peer: peer %s %s",
+			     type_to_string(trc, struct pubkey, &id),
+			     !peer ? "not found" : "not local to this daemon");
```

Comment:
`not local to this daemon` (aka gossipd) is equivalent to `not gossiping`, and may be clearer.